### PR TITLE
docs(entity): add recipe for entity adapters with extra selectors

### LINKED ISSUE #3800: Add a recipe for using entity selectors with extra selectors of a feature
### Information

`extraSelectors` already allows the addition of other selectors to a feature created with `createFeature`, this recipe will expand on using autogenerated selectors from an `EntityAdapter`, how to add them to `extraSelectors` and combine with other selectors in that context

### Documentation page

_No response_

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by danielglejzner**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[X] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

New recipe file for entity adapters with extra selectors

Closes #3800 

## What is the new behavior?
New recipe file for entity adapters with extra selectors

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: projects/ngrx.io/content/guide/entity/recipes/entity-adapters-with-extra-selectors.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+In the component, we `inject` the `Store` service from NgRx and select the data we need from the state using the selectors generated by the `createFeature` function.
+
+<code-example header="userList.component.ts">
+export class UserListComponent{
+	readonly store = inject(Store);
+	readonly isUserSelected$ = this.store.select(userFeature.selectIsUserSelected);
+  readonly selectedUser$ = this.store.select(userFeature.selectSelectedUser);
+}
+</code-example>
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/entity/recipes/entity-adapters-with-extra-selectors.md:84]:
```suggestion
<code-example header="user-list.component.ts">

import { usersFeature } from './users.state';

@Component({ /* ... */ })
export class UserListComponent{
	private readonly store = inject(Store);

  readonly users$ = this.store.select(usersFeature.selectAll);
	readonly isUserSelected$ = this.store.select(usersFeature.selectIsUserSelected);
  readonly selectedUser$ = this.store.select(usersFeature.selectSelectedUser);
}

</code-example>
```

**Review by markostanimirovic — CHANGES_REQUESTED** [review]
Thanks Daniel! I like the order on this page. Few suggestions regarding the examples 👇 

The text should be also adapted according to updated examples.

**Comment by DanielSofomo** [comment]
Great suggestions @markostanimirovic - updated :)

**Comment by danielglejzner** [comment]
Great suggestions @markostanimirovic - updated :)

File: projects/ngrx.io/content/guide/entity/recipes/entity-adapters-with-extra-selectors.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# Using Entity Adapters with Extra Selectors
```
[the remark this answers is not in the document]
Comment (timdeschryver) [reply projects/ngrx.io/content/guide/entity/recipes/entity-adapters-with-extra-selectors.md]: 👍

File: projects/ngrx.io/content/guide/entity/recipes/entity-adapter-with-feature-creator.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
+# Using Entity Adapter with Feature Creator
+
+This example demonstrates a User state management using NgRx Store with `Entity Adapters` and `Extra Selectors` for a feature.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/entity/recipes/entity-adapter-with-feature-creator.md:3]: Thoughts? It's mainly to add links to the adapter and selector docs.

```suggestion
This recipe demonstrates how to combine [entity adapter](/guide/entity/adapter#entity-adapter) (selectors) within [extraSelectors](/guide/store/feature-creators#providing-extra-selectors).
As an example, let's look at some code from a User state management feature.
```

File: projects/ngrx.io/content/guide/entity/recipes/entity-adapter-with-feature-creator.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  source: 'User List Page',
+  events: {
+    addUser: props&lt;{ user: User }&gt;(),
+    selectUser: props&lt;{ userId: string }&gt;(),
+  },
+});
+
+</code-example>
+
+Then use the `createReducer` function from NgRx to create a reducer for the `User` feature.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/entity/recipes/entity-adapter-with-feature-creator.md:43]:
```suggestion
Then use the `createReducer` function to create a reducer for the `User` feature.

```

File: projects/ngrx.io/content/guide/entity/recipes/entity-adapter-with-feature-creator.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  ),
+  on(UserListPageActions.selectUser, (state, { userId }) => ({
+    ...state,
+    selectedUserId: userId,
+  }))
+);
+
+</code-example>
+
+Then create the `User` feature using the `createFeature` function. This will generate all basic selectors automatically, and we can specify extra selectors using the `extraSelectors` option.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/entity/recipes/entity-adapter-with-feature-creator.md:62]:
```suggestion
Then create the `User` feature using the `createFeature` function. 
This generates all the basic selectors automatically, and we can specify extra selectors using the `extraSelectors` option.

```

File: projects/ngrx.io/content/guide/entity/recipes/entity-adapter-with-feature-creator.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+      selectSelectedUserId,
+      selectEntities,
+      (selectedId, entities) => selectedId ? entities[selectedId] : null
+    ),
+  }),
+});
+
+</code-example>
+
+In the component, we `inject` the `Store` service from NgRx and select the data we need from the state using the selectors generated by the `createFeature` function.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/entity/recipes/entity-adapter-with-feature-creator.md:87]:
```suggestion
To use the selector within a component, `inject` the `Store` and select the data from the state using the selectors generated by the `createFeature` function.

```

**Review by timdeschryver — COMMENTED** [review]
What about a "putting it all together" section where the code snippets are all shown within a single file, which makes it easy to copy paste?
Or, we we can also create the file first and insert comments that include a number to reference it later and provide more info/context.

**Review by timdeschryver — COMMENTED** [review]
What about a "putting it all together" section where the code snippets are all shown within a single file, which makes it easy to copy paste?
Or, we we can also create the file first and insert comments that include a number to reference it later and provide more info/context.

File: projects/ngrx.io/content/guide/entity/recipes/entity-adapter-with-feature-creator.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+</code-example>
+
+Then use the `createReducer` function from NgRx to create a reducer for the `User` feature.
+
+<code-example header="users.state.ts">
+
+import { createReducer, on } from '@ngrx/store';
+import { UserListPageActions } from './user-list-page.actions';
+const reducer = createReducer(
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/entity/recipes/entity-adapter-with-feature-creator.md:49]: We can also add an empty line between imports and reducer. The rest of the code snippets look great! :)

```suggestion
import { UserListPageActions } from './user-list-page.actions';

const reducer = createReducer(

```

File: projects/ngrx.io/content/guide/entity/recipes/entity-adapter-with-feature-creator.md (hunk 6 above)
Comment (timdeschryver) [reply projects/ngrx.io/content/guide/entity/recipes/entity-adapter-with-feature-creator.md:87]: Rewording this provide a smoother transition on what the next snippet does.

**Comment by danielglejzner** [comment]
Pushed a batch of changes based on your suggestions :) - ready for re-review.

**Review by timdeschryver — APPROVED** [review]
Thanks! 🤩

**Pull request merged**